aiida/plugins/entry_point.py

```diff
@@ -332,3 +334,27 @@ def is_valid_entry_point_string(entry_point_string):
         return False
 
     return group in entry_point_group_to_module_path_map
+
+
+@functools.lru_cache(maxsize=None)
+def is_registered_entry_point(class_module, class_name, groups=None):
+    """Verify whether the class with the given module and class name is a registered entry point.
+
+    .. note:: this function only checks whether the class has a registered entry point. It does explicitly not verify
+        if the corresponding class is also importable. Use `load_entry_point` for this purpose instead.
+
+    :param class_module: the module of the class
+    :param class_name: the name of the class
+    :param groups: optionally consider only these entry point groups to look for the class
+    :return: boolean, True if the class is a registered entry point, False otherwise.
+    """
+    if groups is None:
+        groups = list(entry_point_group_to_module_path_map.keys())
+
+    registered_entry_points = []
+
+    for group in groups:
+        for entry_point in ENTRYPOINT_MANAGER.iter_entry_points(group):
```

Comment:
Why do we build up all the registered entry points first, instead of just returning `True` as soon as we encounter a matching one (and `False` at the end)? 

Since the total number of entry points is not gigantic it doesn't make much of a difference, though.

Reply:
I have changed it now, but I had the original implementation thinking that this would help with the caching added by the decorator. But to be honest I am not sure how this works. Since we are calling this function now at every `Node.store()` I want to make sure that the overhead is acceptable and we don't impact the performance too much. I will try and run some actual benchmarks.

Reply:
The `functools.lru_cache` just checks if the input arguments are the same, and then returns an existing result (if any). Since `registered_entry_points` was just a local variable inside the function here, that's not preserved in any way.

To cache that part, we would need to put the part setting up the registered entry points into a separate function, and decorate _that_ with `lru_cache`.

Reply:
But that wouldn't matter, right. If we call the function for the same plugin class, so with the same function arguments, it will instantly return the same boolean result, without having to go through the function body again? So essentially that is exactly what we want. I was indeed trying to prevent from having to implement our own caching of the registered entry point mapping, even though it would be not that much work.

Reply:
Yes, I think this is fine - my comment was more a hypothetical, if you _also_ wanted to cache the full list of AiiDA-related entry points.

To be honest I don't think the performance difference to the previous implementation is even that big now, but (to me at least) this seems like the more straightforward way of doing it.